stock_picking_customer_ref/models/sale_order.py

```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
+# Copyright 2015 AvanzOSC - Alfredo de la Fuente
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.multi
+    def write(self, vals):
+        picking_obj = self.env['stock.picking']
+        # The field "client_order_ref" of pickings, is a calculated field. It
+        # can not be put into the field dependence "sale_id".
+        for sale in self:
+            if (vals.get('client_order_ref', False) and
+                    sale.procurement_group_id):
+                cond = [('group_id', '=', sale.procurement_group_id.id)]
+                pickings = picking_obj.search(cond)
+                pickings.write({
```

Comment:
Yes, the problem here seems to be that the compute field in picking is only going to be triggered when the picking is created and the ref already existed. So the aproach should be refocused to either:

- Get rid of the compute field and catch client ref on the picking creation.
- Get rid of the sale methods and try to compute depending on another field (incomplete approach?)

Your opinion is welcome.